### PR TITLE
gce-image: set uniqu image name

### PR DESCRIPTION
**Pull request by yaronkaikov**
got the following error during gce-image build:
```
03:09:46  --> googlecompute: Error creating instance: googleapi: Error 409: The resource 'projects/scylla-images/zones/europe-west1-b/instances/scylla-4-6-dev-0-20210521-a2379b96b1-1' already exists, alreadyExists
```

This is since we are building from same package version and the image
name is not unique.

Let's make sure each gce image buid has a unique name

Fixes: 	https://github.com/scylladb/scylla-pkg/issues/1879

### CODE REVIEW

File: gce/image/scylla_gce.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
       "machine_type": "{{user `instance_type`}}",
       "metadata": {"block-project-ssh-keys": "TRUE"},
       "image_family": "scylla",
-      "image_name": "scylla-{{user `scylla_branch_version`| clean_resource_name}}-build-{{user `scylla_build_id`| clean_resource_name}}",
+      "image_name": "scylla-{{user `scylla_image_name`}}-build-{{user `scylla_build_id`}}",
```
Comment (bentsi) [inline gce/image/scylla_gce.json:14]: why did you remove `clean_resource_name` it 

> clean_resource_name will convert upper cases to lower cases and replace illegal characters with a "-" character. 

https://www.packer.io/docs/templates/legacy_json_templates/engine#clean_resource_name

Comment (yaronkaikov) [reply gce/image/scylla_gce.json]: By mistake. fixed

File: gce/image/build_image.sh [hunk 2]
```diff
@@ -163,6 +162,7 @@ else
 
 fi
 
+SCYLLA_IMAGE_NAME=$(echo $SCYLLA_VERSION | sed "s/\./-/g")
```
Comment (bentsi) [inline gce/image/build_image.sh:165]: are you sure you need the sed part? `clean_resource_name` will normalize the name..

Comment (bentsi) [reply gce/image/build_image.sh:165]: and better to put $(...) in quotes

Comment (yaronkaikov) [reply gce/image/build_image.sh:165]: Let me verify it

Comment (yaronkaikov) [reply gce/image/build_image.sh]: @bentsi you were right. it's not needed. removed

File: gce/image/build_image.sh [hunk 3]
```diff
@@ -18,7 +18,6 @@ DRY_RUN=false
 source ../../SCYLLA-VERSION-GEN
 
 PRODUCT=$(cat build/SCYLLA-PRODUCT-FILE)
-BUILD_ID=$(date -u '+%FT%H-%M-%S')
```
Comment (bentsi) [inline gce/image/build_image.sh:21]: if you delete the build_id how it will be passed in line 195?

Comment (yaronkaikov) [reply gce/image/build_image.sh:21]: we provide it as a parameter

Comment (yaronkaikov) [reply gce/image/build_image.sh:21]: I can set maybe a default value like - `000` or something

Comment (bentsi) [reply gce/image/build_image.sh:21]: let's keep it

**Review by bentsi — APPROVED** [review]
